packages/flutter/lib/src/rendering/grid.dart

```diff
@@ -73,65 +273,78 @@ class RenderGrid extends RenderBox with ContainerRenderObjectMixin<RenderBox, Gr
       child.parentData = new GridParentData();
   }
 
+  Size _getIntrinsicSize(BoxConstraints constraints) {
+    return constraints.constrain(_delegate.getGridSpecification(constraints, childCount).gridSize);
+  }
+
   double getMinIntrinsicWidth(BoxConstraints constraints) {
     assert(constraints.isNormalized);
-    // We can render at any width.
-    return constraints.constrainWidth(0.0);
+    return _getIntrinsicSize(constraints).width;
   }
 
   double getMaxIntrinsicWidth(BoxConstraints constraints) {
     assert(constraints.isNormalized);
-    double maxWidth = childCount * _maxChildExtent;
-    return constraints.constrainWidth(maxWidth);
+    return _getIntrinsicSize(constraints).width;
```

Comment:
This is wrong. It should return the smallest width allowed within the constraints that would not increase the height. As a specific example, consider what the output will be if the input is unconstrained. It should not be infinite.